Mathlib/Tactic/Linter/UnusedSetOptionIn.lean

```diff
@@ -0,0 +1,65 @@
+/-
+Copyright (c) 2024 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+
+import Mathlib.Tactic.Lemma
+import Lean.Elab.Command
+import Lean.Linter.Util
+
+/-!
+The `unusedSetOptionIn` linter warns against "most external" unused `set_option ... in`s.
+-/
+
+open Lean Elab Command
+
+/-- converts
+* `theorem x ...` to  `some (example ... , x)`,
+* `lemma x ...`   to  `some (example ... , x)`,
+* `example ...`   to ``some (example ... , `example)``,
+*  everything else goes to `none`.
+-/
+def toExample {m : Type → Type} [Monad m] [MonadRef m] [MonadQuotation m] :
+    Syntax → m (Option (Syntax × Syntax))
+  | `($dm:declModifiers theorem $did:declId $ds* : $t $dv:declVal) => do
+    return some (← `($dm:declModifiers example $ds* : $t $dv:declVal), did.raw[0])
+  | `($dm:declModifiers lemma $did:declId $ds* : $t $dv:declVal) => do
+    return some (← `($dm:declModifiers example $ds* : $t $dv:declVal), did.raw[0])
+  | `($dm:declModifiers example $ds:optDeclSig $dv:declVal) => do
+    return some (← `($dm:declModifiers example $ds $dv:declVal), mkIdent `example)
+  | _ => return none
+
+/-- The `unusedSetOptionIn` linter warns against "most external" unused `set_option ... in`s. -/
```

Comment:
... possibly with the same phrasing

Reply:
Yes, I should have recycled this using `inherit_doc`.